crates/cdk/src/mint_url.rs

```diff
@@ -16,20 +16,48 @@ pub enum Error {
     /// Url error
     #[error(transparent)]
     Url(#[from] ParseError),
+    /// Invalid URL structure
+    #[error("Invalid URL")]
+    InvalidUrl,
 }
 
 /// MintUrl Url
 #[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize)]
 pub struct MintUrl(String);
 
 impl MintUrl {
-    /// New mint url
-    pub fn new<S>(url: S) -> Self
-    where
-        S: Into<String>,
-    {
-        let url: String = url.into();
-        Self(url.trim_end_matches('/').to_string())
+    fn format_url(url: &str) -> Result<String, Error> {
+        if url.is_empty() {
+            return Ok(String::new());
+        }
```

Comment:
Should this return an error instead of an empty string?

Reply:
The test in `mod_rotate_keyset` passes an empty string, so this part just reproduces the old code. What do you think?

Reply:
I think it should return an error. We can change the test MintConfg to have a valid url with: 
```rust

        let config = MintConfig::<'_> {
            mint_url: "http://example.com",
            ..Default::default()
        };
```

https://github.com/cashubtc/cdk/blob/1cd80deaca28a63094931020bd90a26d9300998d/crates/cdk/src/mint/mod.rs#L1572

like we do in the test above it:

https://github.com/cashubtc/cdk/blob/1cd80deaca28a63094931020bd90a26d9300998d/crates/cdk/src/mint/mod.rs#L1532-L1536